oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java

```diff
@@ -284,12 +407,57 @@ private void traverseChildren(NodeState state, String path, ReferenceCollector c
                     for (String node: nodeList) {
                         state = state.getChildNode(node);
                     }
-                    traverseChildren(state, path, collector);
+
+                    if (inclusionRegex == null || inclusionRegex.length == 0) {
+                        traverseChildren(state, path, collector);
+                    } else {
+                        for (String regex : inclusionRegex) {
+                            Map<NodeState, String> inclusionMap = new HashMap<NodeState, String>();
+                            getInclusionListFromRegex(state, path, regex, inclusionMap);
+                            if (inclusionMap.size() == 0 ) {
+                                System.out.println("No Valid paths found for traversal, " +
+                                        "for the inclusion Regex " + regex + " under the path " + path);
+                                continue;
+                            }
+                            for(NodeState s : inclusionMap.keySet()) {
+                                traverseChildren(s, inclusionMap.get(s), collector);
+                            }
+                        }
+                    }
+
                 }
             }
 
 
         }
+
+        private void getInclusionListFromRegex(NodeState rootState, String rootPath, String inclusionRegex, Map<NodeState, String> inclusionNodeStates) {
+            Splitter delimSplitter = Splitter.on("/").trimResults().omitEmptyStrings();
+            List<String> pathElementList = delimSplitter.splitToList(inclusionRegex);
+
+            Joiner delimJoiner = Joiner.on("/").skipNulls();
+
+            // Get the first pathElement from the regexPath
+            String pathElement = pathElementList.get(0);
+            // If the pathElement == *, get all child nodes and scan under them for the rest of the regex
+            if (pathElement.equals("*")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/gh/oak-indexing/jackrabbit-oak/pullRequest?prid=5079914)